rdagent/utils/agent/tpl.py

```diff
@@ -43,11 +44,19 @@ def load_content(uri: str, caller_dir: Path | None = None, ftype: str = "yaml")
     # load file_path with priorities.
     if path_part.startswith("."):
         file_path_l = [caller_dir / f"{path_part[1:].replace('.', '/')}.{ftype}"]
+        if RD_AGENT_SETTINGS.app_tpl is not None:
+            file_path_l.insert(0, PROJ_PATH / RD_AGENT_SETTINGS.app_tpl / file_path_l[0].relative_to(PROJ_PATH))
     else:
         file_path_l = [
             Path(path_part.replace(".", "/")).with_suffix(f".{ftype}"),
             (PROJ_PATH / path_part.replace(".", "/")).with_suffix(f".{ftype}"),
         ]
+        # NOTE: for application's template to override the default template
+        if RD_AGENT_SETTINGS.app_tpl is not None:
+            file_path_l.insert(
+                0, (PROJ_PATH / RD_AGENT_SETTINGS.app_tpl / path_part.replace(".", "/")).with_suffix(f".{ftype}")
+            )
+            file_path_l.insert(0, (PROJ_PATH.parent / path_part.replace(".", "/")).with_suffix(f".{ftype}"))
```

Comment:
```suggestion
            # NOTE: when we can both load tpl from tpl; to avoid recursive extension. 
            # e.g. we want app_tpl/a.b.c extend rdagent/a.b.c;  so we allow specifying in a upper
            #        level. for example,  rdagent.a.b.c;
            file_path_l.insert(0, (PROJ_PATH.parent / path_part.replace(".", "/")).with_suffix(f".{ftype}"))
```